lib/ansible/playbook/role/__init__.py

```diff
@@ -98,19 +98,30 @@ def hash_params(params):
 
 class Role(Base, Conditional, Taggable, CollectionSearch, Delegatable):
 
-    def __init__(self, play=None, from_files=None, from_include=False, validate=True, public=True):
+    def __init__(self, play=None, from_files=None, from_include=False, validate=True, public=None, static=True):
```

Comment:
I don't think play roles [have a way to set public](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/play.py#L241), so this will make them use config (new default) with no way to reinstate the previous behavior.

Reply:
`include_role` has a `public` property, which i've tested (need to add to tests here)

Reply:
ah, yes, `roles:` has no way to override, but this is true for any other parameter, i.e `tasks_from:`

Reply:
and the default was always meant to affect them

Reply:
> and the config was always meant to affect them

wasn't objecting to the behavior, just pointing out there's no way to reinstate the original behavior

Reply:
only with the global config, this was always intended when that config was added